packages/shared/src/components/notifications/InAppNotification.tsx

```diff
@@ -79,15 +85,26 @@ export function InAppNotificationElement(): ReactElement {
     };
   }, [payload]);
 
-  const onNotificationClick = (id: string, type: NotificationType) => {
+  const onNotificationClick = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    id: string,
+    type: NotificationType,
+  ) => {
     trackEvent({
       event_name: AnalyticsEvent.ClickNotification,
       target_id: id,
       extra: JSON.stringify({ origin: Origin.RealTime, type }),
     });
+
+    if (type === NotificationType.SquadSubscribeNotification) {
+      event.preventDefault();
+      onTogglePermission(NotificationPromptSource.NotificationItem);
+    }
```

Comment:
I would actually handle all of this on the `?notification` part as it has nothing to do with this.

Reply:
The issue is, the link is based on the BE. You could be at the notifications page or on the Squads page seeing the in app notification, and when clicking, the URL gives us a complete link, not just adding a parameter to the current query.

Reply:
Yeah exactly so normally we use the main page for all of these.
As mentioned in the DR we decided on using the same approach for squad creation which is the homepage.
I would stick to what was decided there.

Reply:
Yeah but the DR dictates a **similar** approach not necessarily the same. But if it is not really important to consider the UX then I'm all for redirection.